parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordWriter.java

```diff
@@ -112,6 +153,10 @@ public ParquetRecordWriter(
   @Override
   public void close(TaskAttemptContext context) throws IOException, InterruptedException {
     internalWriter.close();
+    // release after the writer closes in case it is used for a last flush
+    if (codecFactory != null) {
```

Comment:
Yeah, I wrapped this in a try/finally just to be safe as well. I'm just rebasing on the PARQUET-99 fix and I'll post a new copy.